test/spec/services/user.js

```diff
@@ -5,78 +5,122 @@ describe('Service: user', function () {
+        resolve({
+          data: testUsers
+        });
+      } else {
+        reject({
+          statusText: "Could not complete get"
+        });
+      }
+    });
+  };
+
+  var fakePostPromise = function (shouldResolve) {
+    return q(function (resolve, reject) {
+      if (shouldResolve){
+        resolve({
+          status: 300
+        });
+      } else {
+        reject({
+          statusText: "Could not complete post"
+        });
+      }
+    });
+  };
+
   it('should do something', function () {
     expect(!!user).toBe(true);
   });
 
-  it('should succeed when registering a valid username', function() {
-    var promise = user.attemptRegister('John');
+  it('should succeed when registering an unused username', function() {
+    var newUser = "john";
+    spyOn(http, "get").and.returnValue(fakeGetPromise(true));
+    spyOn(http, "post").and.returnValue(fakePostPromise(true));
+    var promise = user.register(newUser);
     var spy = jasmine.createSpy('add user spy');
     promise.then(spy);
     scope.$apply();
 
-    expect(spy).toHaveBeenCalledWith('John');
+    expect(http.get).toHaveBeenCalled();
+    expect(http.post).toHaveBeenCalled();
+    expect(spy).toHaveBeenCalledWith(newUser);
   });
 
   it('should fail when registering existing username', function() {
-    user.attemptRegister('John');
-    var promise = user.attemptRegister('John');
-    var spy = jasmine.createSpy('duplicate user spy');
-    promise.then(angular.noop, spy);
+    var existingUser = "testUser";
+    spyOn(http, "get").and.returnValue(fakeGetPromise(true));
+    spyOn(http, "post").and.returnValue(fakePostPromise(true));
+    user.register(existingUser);
+    var promise = user.register(existingUser);
+    var failSpy = jasmine.createSpy('duplicate user spy');
+    promise.then(angular.noop, failSpy);
     scope.$apply();
 
-    expect(spy).toHaveBeenCalled();
+    expect(http.get).toHaveBeenCalled();
+    expect(http.post).not.toHaveBeenCalled();
+    expect(failSpy).toHaveBeenCalled();
   });
 
   it('should succeed when registering two different usernames', function() {
-    user.attemptRegister('John');
-    var promise = user.attemptRegister('Mary');
-    var spy = jasmine.createSpy('different users spy');
-    promise.then(spy);
-    scope.$apply();
-
-    expect(spy).toHaveBeenCalledWith('Mary');
-  });
-
-  it('should fail when registering two different usernames with the first username repeated afterwards', function() {
-    user.attemptRegister('John');
-    user.attemptRegister('Alice');
-    var promise = user.attemptRegister('John');
-    var spy = jasmine.createSpy('different users spy');
-    promise.then(angular.noop, spy);
+    var name1 = "will";
+    var name2 = "monkey";
+    spyOn(http, "get").and.returnValue(fakeGetPromise(true));
+    spyOn(http, "post").and.returnValue(fakePostPromise(true));
+    var successSpy = jasmine.createSpy('succeed spy');
+    var failSpy = jasmine.createSpy('fail spy');
```

Comment:
There does not seem to be any use for this spy. Just remove it if you are not using it.